internal/fields/validate.go

```diff
@@ -320,6 +320,11 @@ func (v *Validator) validateDocumentValues(body common.MapStr) multierror.Error
 				continue
 			}
 
+			// skip checking `data_stream.dataset` because reroute processor will change this field value
```

Comment:
Note that the reroute processor also changes `event.dataset` so that it matches `data_stream.dataset`.